docs/faq/code-analysis/which-metrics-does-codacy-calculate.md

```diff
@@ -0,0 +1,114 @@
+---
+description: Codacy scans your code for issues and calculates the metrics code complexity, duplication, and coverage. Besides this, Codacy also calculates a grade for your repository and files based on all calculated code quality metrics.
+---
+
+# Which metrics does Codacy calculate?
+
+Codacy performs static code analysis and calculates code duplication, code complexity, and code coverage metrics for [most supported programming languages](../../getting-started/supported-languages-and-tools.md).
+
+The following sections describe how Codacy calculates each supported metric and where you can see each metric on the Codacy UI:
+
+-   [Grade](#grade)
+-   [Issues](#issues)
+-   [Complexity](#complexity)
+-   [Duplication](#duplication)
+-   [Code coverage](#code-coverage)
+
+## Grade
+
+Codacy assigns an overall grade to your repository branches and to individual files to help you understand the code quality of your repository. Grades depend on a score that takes into account a weighted average of all supported code quality metrics, and range from **A** (highest grade) to **F** (lowest grade):
```

Comment:
This doesn't explain the exact formula that we use to calculate the score that is then transformed into a grade letter.

Reply:
@prcr "help you understand" here is misleading. Consider "assess the code quality".

Reply:
@prcr Didn't know where to include this comment:
There also a "null" grade, that is grey and with a ' ! '. Might be good to add what it means in the docs.

Reply:
Good suggestions, I applied [these changes](https://github.com/codacy/docs/pull/1104/commits/0b2ece5f0a27c004e692211c09f42a54518bc21a).

Reply:
> There also a "null" grade, that is grey and with a ' ! '. Might be good to add what it means in the docs.

After feedback from @IGFCoimbra, we decided to remove the reference to the case where the grade isn't calculated because of an exception, since this page is focused on the definition of the metrics. We can still mention the greyed out `!` grade on a troubleshooting page like [this one](https://docs.codacy.com/faq/repositories/how-do-i-reanalyze-my-repository/).